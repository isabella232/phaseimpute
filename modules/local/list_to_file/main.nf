process LIST_TO_FILE {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gawk:5.3.0' :
        'biocontainers/gawk:5.3.0' }"

    input:
    tuple val(meta), path(input, arity: '0..*'), val(id)

    output:
    tuple val(meta), path('*.id.txt'), path('*.noid.txt'), emit: txt
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    # Take all files of the input and list them in a file
    # and add as second column the id
    awk 'BEGIN {
        split("${input}", f, " ");
        ids = "${id}";
        gsub(/[\\[\\]]/, "", ids);
        split(ids, i, ", ");
        for (j in f) print f[j], i[j]
    }' > ${prefix}.id.txt

    # Take all files of the input and list them in a file
    # without the id

    awk 'BEGIN {
        split("${input}", f, " ");
        for (j in f) print f[j]
    }' > ${prefix}.noid.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gawk: \$(awk -Wversion | sed '1!d; s/.*Awk //; s/,.*//')
    END_VERSIONS
    """
}
