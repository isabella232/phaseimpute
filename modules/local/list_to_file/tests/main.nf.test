nextflow_process {

    name "Test Process LIST_TO_FILE"
    script "../main.nf"
    process "LIST_TO_FILE"

    tag "modules"
    tag "modules_local"
    tag "list_to_file"
    tag "gawk"

    test("Aggregate all files into a list") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [id: "all"],
                    [file("file1.txt"), file("file2.txt")],
                    ["A", "B"]
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert snapshot(path(process.out.txt[0][1]).readLines()).match("Content_withid") },
                { assert snapshot(path(process.out.txt[0][2]).readLines()).match("Content_noid") }
            )
        }
    }
}
