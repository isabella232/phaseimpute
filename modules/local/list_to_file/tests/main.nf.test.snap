{
    "Aggregate all files into a list": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all"
                        },
                        "all.id.txt:md5,6ffc4e703a69bb652dc666a8dcae1863",
                        "all.noid.txt:md5,045a6e673d95a6e8152a1d06959db93a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ab51934043f45af87e4348a704066b0f"
                ],
                "txt": [
                    [
                        {
                            "id": "all"
                        },
                        "all.id.txt:md5,6ffc4e703a69bb652dc666a8dcae1863",
                        "all.noid.txt:md5,045a6e673d95a6e8152a1d06959db93a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ab51934043f45af87e4348a704066b0f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-11-06T13:55:57.105169612"
    },
    "Content_withid": {
        "content": [
            [
                "file1.txt A",
                "file2.txt B"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-16T15:51:19.289182489"
    },
    "Content_noid": {
        "content": [
            [
                "file1.txt",
                "file2.txt"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-16T16:15:18.915373799"
    }
}